clients/csharp-uwp/docs/sample-app-walkthrough.md

```diff
@@ -0,0 +1,106 @@
+#### Pass input audio to Direct Line Speech
+DialogManager completes step 8 by initializing the AgentAudioInputProvider and passing it to an instance of [DirectLineSpeechDialogBackend](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/blob/master/clients/csharp-uwp/UWPVoiceAssistantSample/DirectLineSpeechDialogBackend.cs) through DirectLineSpeechDialogBackend.SetAudioSource. The DirectLineSpeechDialogBackend class manages the Speech SDK classes used to interface with Direct Line Speech. 
+#### Initiate keyword verification
+When the AgentAudioInputProvider fires the DataAvailable event, the bytes from the activation audio are [passed to the Speech SDK](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/blob/master/clients/csharp-uwp/UWPVoiceAssistantSample/DirectLineSpeechDialogBackend.cs#L192) and analyzed by the local 2nd stage keyword detector.
+#### Handle 2nd stage keyword verification
+When the 2nd stage keyword detector detects a keyword, the DialogServiceConnector object in DirectLineSpeechDialogBackend will fire the SpeechRecognizing event with the result reason ResultReason.RecognizingKeyword. Note: there is no "keyword rejected" signal from the DialogServiceConnector for 2nd stage verification, so there must be a manually implemented timeout to cancel verification in case of failure. In the sample application, this "rejection timer" is implemented in [SignalDetectionHelper](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/blob/master/clients/csharp-uwp/UWPVoiceAssistantSample/SignalDetectionHelper.cs).
+#### Handle 3rd stage keyword verification
+On successful 2nd stage keyword verification, the DialogServiceConnector automatically sends the audio to the cloud for 3rd stage keyword verification. When it verifies the keyword with cloud verification, the DialogServiceConnector object will fire the SpeechRecognized event with the result reason ResultReason.RecognizedKeyword. If cloud verification fails, the result reason will be ResultReason.NoMatch. In either case, the DialogManager surfaces these events as SignalConfirmed and SignalRejected.
+
+### 10. If the keyword is verified, continue in the background or move to the foreground to display UI
+The app is running in the background after it is activated by MVA, so it cannot display UI. For an audio-only app interaction, your application can simply stay in the background, play audio output, and listen to audio input. To display UI, call ConversationalAgentSession.RequestForegroundActivationAsync. In the sample app, this is called in [App.xaml.cs](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/blob/master/clients/csharp-uwp/UWPVoiceAssistantSample/App.xaml.cs)'s OnSignalConfirmed which is triggered by the SignalConfirmed event in DialogManager.
```

Comment:
This one is an interesting one: it's true, but I've wondered when and how this would be used in practice. At the time of activation, we don't know what the user is going to say; that makes foregoing the foreground transition pretty ugly if interaction was expected. Maybe if the screen is off? But then that could be bad, too! What do you think?

Reply:
I figured we had requestforeground as a separate method because showing UI was optional. If we recommend all apps show UI, then I'll remove the piece about running in the background.